### PR TITLE
Working with java synchronization

### LINKED ISSUE #5936: Working with java synchronization
#### Proposed title of article
[Languages] Working with java synchronization

#### Proposed article introduction
The synchronization mechanism in java prohibits multiple threads from accessing shared resources. Shared resources include external files, class variables, and database information.
Synchronization is prevalent in multithreaded programming. Synchronized is the keyword that allows your code to run on a single thread at a time without interruption.

#### Key takeaways 
After reading this article, the reader should be able to understand:
-  What is synchronization in java and why do we need it.
- How synchronization works
- How to declare a method as synchronized
- Synchronized method vs Synchronized block 

#### Article quality
This will be an in-depth article covering all the aspects of this concept. It will explain key concepts to readers which is unavailable in the documentations and other platforms. I aim at making it more beginner-friendly simple and easy to understand. More on the coverage will be on the types of synchronization as well as multi-threading with and without synchronization

#### References
N/A

### PR DESCRIPTION
**Pull request by gracy20**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5936 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

File: content/articles/working-with-java-synchronization/index.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+To understand the two, let us first look at each separately:
+
+#### Synchronized methods
+These consist of the following properties:
+
+- A synchronized method locks down the whole object. No other thread may access synchronized methods in the object while the method is executing.
+- In non-static methods, this locks them whereas static methods are locked by their class.
+- Synchronized methods need locks on the current object or the class level lock if they are static. This is because the lock is obtained when the thread enters and released when the thread quits (naturally or by throwing an Exception).
+- Synchronized methods keep the lock for the method scope.
+- A synchronized static function locks the class object.
```
Comment (john-amiscaray) [inline content/articles/working-with-java-synchronization/index.md:142]: Not sure what you mean by class object

File: content/articles/working-with-java-synchronization/index.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+- A synchronized method locks down the whole object. No other thread may access synchronized methods in the object while the method is executing. In non-static methods, this locks them whereas static methods are locked by their class.
+- Synchronized methods need locks on the current object or, in the case of static methods, the whole class. This is because the lock is obtained when the thread enters and released when the thread quits (naturally or by throwing an Exception).
+- Synchronized methods keep the lock for the method scope.
+- A synchronized static function locks the class object.
+
+#### Synchronized blocks
+The following is all that it entails:
+
+- This keyword locks synced blocks on the item between parentheses. Locked objects are only accessible to the synchronized block's threads.
```
Comment (john-amiscaray) [inline content/articles/working-with-java-synchronization/index.md]: Not clear exactly what you mean by:
 
*on the item between parentheses*

Do you mean the code within the block is locked?

File: content/articles/working-with-java-synchronization/index.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+- Synchronized blocks utilize it as a lock.
```
Comment (john-amiscaray) [inline content/articles/working-with-java-synchronization/index.md:148]: What are you referring to as *it* here?

**Review by john-amiscaray — CHANGES_REQUESTED** [review]
Please have a look at my feedback and implement my suggested revisions before we carry on with this review:

**Comment by john-amiscaray** [comment]
I will be closing this pull request while you implement my changes. When you are done, please make a new pull request with the changes and ping me in the comments of that new pull request. If you need assistance, feel free to message me via slack.

**Pull request closed without merging**